data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/DynamicIndexManager.java

```diff
@@ -47,9 +46,8 @@ public DynamicIndexManager(final IndexType indexType,
         this.restHighLevelClient = restHighLevelClient;
         this.openSearchSinkConfiguration = openSearchSinkConfiguration;
         this.clusterSettingsParser = clusterSettingsParser;
-        CacheBuilder<String, IndexManager> cacheBuilder = CacheBuilder.newBuilder()
+        Caffeine<String, IndexManager> cacheBuilder = Caffeine.newBuilder()
                         .recordStats()
-                        .concurrencyLevel(1)
```

Comment:
Is there a similar setting in Caffeine that we should make use of here?

Reply:
Caffeine provides another mechanism for achieving optimal performance in concurrent execution scenarios out-of-the-box, so that option is redundant.

Reply:
This was probably `1` because Guava splits the have into N lru segments, where each segment has `maximumWeight / N` capacity. Since you are using weights, that might have meant a large entry was immediately evicted so the concurrency level was reduced to compensate. Since this mostly impacts write throughput it may have been fine for performance, as reads are lock-free. Long story for why this was done (I disagreed) and Caffeine does not need this or have that problem.

Reply:
@ben-manes  thanks for clarifying the specifics!

Reply:
@ben-manes , Thank you for the explanation.